fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/cursors/SizeStatisticsCollectorCursor.java

```diff
@@ -217,32 +222,30 @@ private SizeStatisticsCollectorCursorContinuation(RecordCursorResult<KeyValue> c
                     return RecordCursorProto.SizeStatisticsContinuation.newBuilder().setPartialResults(this.sizeStatisticsResults.toProto()).setContinuation(ByteString.copyFrom(b)).build();
                 } else {
                     //nothing more to aggregate
-                    return RecordCursorProto.SizeStatisticsContinuation.newBuilder().setContinuation(ByteString.copyFrom(RecordCursorEndContinuation.END.toBytes())).build();
+                    return RecordCursorProto.SizeStatisticsContinuation.newBuilder().build();
                 }
             };
         }
 
         @Nullable
         @Override
         public byte[] toBytes() {
-            if (isEnd()) {
-                return null;
-            } else {
-                //form bytes exactly once
-                if (this.cachedBytes == null) {
-                    this.cachedBytes = continuationFunction.apply(this.currentKvResult.getContinuation().toBytes()).toByteArray();
-                }
-                return cachedBytes;
+            //form bytes exactly once
+            if (this.cachedBytes == null) {
+                this.cachedBytes = continuationFunction.apply(this.currentKvResult.getContinuation().toBytes()).toByteArray();
             }
+            return cachedBytes;
         }
 
         @Override
         @Nonnull
+        //instance of this class do not are NOT meant to represent end continuations
```

Comment:
I think putting comments between `@Nonnull` and the function body confuses the annotation processor. Can this be either before `@Override` or in the function body itself? (Also, probably a space between `//` and `instance` (also `instances` instead of `instance`).